### PR TITLE
create category search part for registration

### PR DESCRIPTION
**Pull request by sargiss987**
Have been added category search part for student registration. Now if have been chosen some categories they will be saved via registration. Searching includes categories and subcategories, if no title is written then by default will be fetched all categories and subcategories...

### CODE REVIEW

File: os-server/src/main/java/app/openschool/coursemanagement/CategoryRepository.java [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+import java.util.List;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface CategoryRepository extends JpaRepository<Category, Long> {
+public interface CategoryRepository extends JpaRepository<Category, Integer> {
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/coursemanagement/CategoryRepository.java:12]: Why this is changed to Integer ?

File: os-server/src/main/java/app/openschool/coursemanagement/CourseController.java [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
 import org.springframework.data.domain.Pageable;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @RequestMapping("/api/v1")
+@CrossOrigin("http://localhost:3000")
```
Comment (SevakMart) [inline os-server/src/main/java/app/openschool/coursemanagement/CourseController.java:18]: we need to think, how to make it work only for development environment

File: os-server/src/main/java/app/openschool/coursemanagement/CategoryRepository.java (hunk 1 above)
Comment (sargiss987) [reply os-server/src/main/java/app/openschool/coursemanagement/CategoryRepository.java:12]: The id type of category in DB is INT. I thought probably it is redundant to use Long type in app model.

File: os-server/src/main/resources/db/migration/h2/V1_0_2__update.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+-- -----------------------------------------------------
```
Comment (SevakMart) [inline os-server/src/main/resources/db/migration/h2/V1_0_2__update.sql:1]: why this is added in this PR ?

Comment (sargiss987) [reply os-server/src/main/resources/db/migration/h2/V1_0_2__update.sql:1]: actually, it isn't relevant to this task, but as Hakob and I are working together on DB I've added this one too. It's for his needs.

**Pull request merged**